pkgs/by-name/xw/xwayland-satellite/package.nix

```
@@ -0,0 +1,59 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, clang
+, xwayland
+, xorg
+, xcb-util-cursor
+, libxcb
+, llvmPackages
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "xwayland-satellite";
+  version = "0.4";
+
+  LIBCLANG_PATH = "${llvmPackages.libclang.lib}/lib";
+
+  src = fetchFromGitHub {
+    owner = "Supreeeme";
+    repo = "xwayland-satellite";
+    rev = "v${version}";
+    hash = "sha256-dwF9nI54a6Fo9XU5s4qmvMXSgCid3YQVGxch00qEMvI=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    clang
+  ];
+
+  runtimeDependencies = [
+    xwayland
+  ];
+
+  buildInputs = [
+    xwayland
+    libxcb
+    xcb-util-cursor
+  ];
+
+  # Highly impure tests, which fail to run in the sandbox
+  doCheck = false;
+  # checkFlags = [
+  #   "--skip=xwayland-satellite::tests::integration"
+  # ];
```

Only the actual integration tests need to be disabled:
```suggestion
  # disable Xwayland integration tests which need a running display server 
  checkFlags = [
    "--exact"
    "--skip=copy_from_wayland"
    "--skip=copy_from_x11"
    "--skip=input_focus"
    "--skip=quick_delete"
    "--skip=reparent"
    "--skip=toplevel_flow"
  ];
```